src/main/java/com/google/devtools/build/lib/analysis/skylark/SkylarkRuleClassFunctions.java

```diff
@@ -398,8 +398,13 @@ public BaseFunction rule(
               execCompatibleWith.getContents(String.class, "exec_compatile_with"),
               ast.getLocation()));
     }
-    if (executionPlatformConstraintsAllowed) {
+
+    if (funcallEnv.getSemantics().incompatibleDisallowRuleExecutionPlatformConstraintsAllowed()) {
```

Comment:
This doesn't look like the fully incompatible change you want -- a user can still specify rule(..., execution_platform_constraints_allowed = False), it just would have no effect.

I'd recommend the following instead of any change in this class:

1. Introduce a new FlagIdentifier entry in StarlarkSemantics.java for the new flag
2. In SkylarkRuleFunctionsApi.java, find the @Param for execution_platform_constraints_allowed, and set the following fields on it:
     A. disableWithFlag = <new flag identifier>
     B. valueWhenDisabled = "True"

This should work out-of-the-box (but you should write a specific test for it, just in case :)) Let me know if you run into any issues!

Reply:
Thanks, this is much cleaner.

Reply:
Don't forget to add a test~!